halo2_proofs/src/plonk/vanishing/prover.rs

```diff
@@ -47,10 +49,25 @@ impl<C: CurveAffine> Argument<C> {
         transcript: &mut T,
     ) -> Result<Committed<C>, Error> {
         // Sample a random polynomial of degree n - 1
-        let mut random_poly = domain.empty_coeff();
-        for coeff in random_poly.iter_mut() {
-            *coeff = C::Scalar::random(&mut rng);
-        }
+        let n_threads = std::thread::available_parallelism().unwrap().get();
```

Comment:
[rayon](https://docs.rs/rayon/latest/rayon/index.html)::[current_num_threads](https://docs.rs/rayon/latest/rayon/fn.current_num_threads.html#) may be more consistent? Not sure if there's any difference in how it counts available threads.